NitroxPatcher/Patches/Dynamic/DevConsole_Update_Patch.cs

```diff
@@ -1,19 +1,56 @@
-﻿using HarmonyLib;
+﻿using System.Collections.Generic;
+using System.Reflection;
+using HarmonyLib;
 using NitroxClient.GameLogic;
 using NitroxModel.Helper;
+using NitroxPatcher.PatternMatching;
+using UnityEngine;
+using static System.Reflection.Emit.OpCodes;
 
 namespace NitroxPatcher.Patches.Dynamic
 {
-    class DevConsole_Update_Patch : NitroxPatch, IDynamicPatch
+    internal class DevConsole_Update_Patch : NitroxPatch, IDynamicPatch
     {
+        /// <summary>
+        ///     Pattern to keep DevConsole disabled when enter is pressed.
+        /// </summary>
+        private static readonly InstructionsPattern devConsoleSetStateTruePattern = new()
+        {
+            InstructionPattern.Call(nameof(Input), nameof(Input.GetKeyDown)),
+            Brfalse,
+            Ldarg_0,
+            Ldfld,
+            Brtrue,
+            Ldarg_0,
+            { Ldc_I4_1, "TrueArgument" },
+            InstructionPattern.Call(nameof(DevConsole), nameof(DevConsole.SetState))
+        };
+
+        public static readonly MethodInfo TargetMethod = Reflect.Method((DevConsole t) => t.Update());
+
+        /// <summary>
+        ///     Prevent console from opening if server wants it disabled.
+        /// </summary>
         public static void Prefix()
         {
             PlatformUtils.SetDevToolsEnabled(!NitroxConsole.DisableConsole);
         }
 
+        public static IEnumerable<CodeInstruction> Transpiler(MethodBase original, IEnumerable<CodeInstruction> instructions) => instructions
+            .Transform(devConsoleSetStateTruePattern, (label, instruction) =>
+            {
+                switch (label)
+                {
+                    case "TrueArgument":
+                        instruction.opcode = Ldc_I4_0;
+                        break;
+                }
+            });
+
         public override void Patch(Harmony harmony)
         {
-            PatchPrefix(harmony, Reflect.Method((DevConsole t) => t.Update()));
+            PatchPrefix(harmony, TargetMethod);
```

Comment:
Do we need to keep a prefix method since we're transpiling the method ?

Reply:
I kept it because prefix is easier to add new code. You prefer I include the prefix change into the transpiler for performance reasons?

Reply:
If we can avoid dynamic patches on Update method, it definitely sounds better.

NitroxConsole.Disable is only set once during session policies exchange. So we don't really need to be reactive on it's value. Unless we want to allow dynamic session policies, but right now it requires a server restart.